### PR TITLE
Addressed some issues in the tutorial mentioned in discussion #22233

### PR DESCRIPTION
**Pull request by MattTriano**
While working through the tutorial, some parts seemed a bit unclear or incorrect, which took me to issue #18950 ( converted into discussion #22233 about 15 minutes ago). I had already resolved some of the open concerns and saw potiuk encouraging people to improve the documentation, so I took it upon myself.

Main Changes:
* Wrote tasks to handle db-table-creation in the DAG rather than leave that as an exercise for the reader.
* Added a step to ensure the existence of the data output directory in the `get_data()` task.
* Fixed the logic in the `merge_data()` task so that its behavior matches the stated description.

---
**^ Add meaningful description above**

Read the **[Pull Request Guidelines](https://github.com/apache/airflow/blob/main/CONTRIBUTING.rst#pull-request-guidelines)** for more information.
In case of fundamental code change, Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvements+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backwards incompatible changes please leave a note in [UPDATING.md](https://github.com/apache/airflow/blob/main/UPDATING.md).

### CODE REVIEW

File: docs/apache-airflow/tutorial.rst [hunk 1]
```diff
@@ -378,62 +378,83 @@ Lets look at another example; we need to get some data from a file which is host
...
-We are now ready write the DAG.
+  create_employees_table = PostgresOperator(
+      task_id="create_employees_table",
+      postgres_conn_id="tutorial_pg_conn",
+      sql="sql/employees_schema.sql"
+  )
 
+and repeat for the ``employees_temp`` table.
 
+Data Retrival Task
```
Comment (ephraimbuddy) [inline docs/apache-airflow/tutorial.rst]:
```suggestion
Data Retrieval Task
```

**Comment by MattTriano** [comment]
I spent a bit more time refactoring (after noticing a thought I left incomplete). I assume most tutorial readers are either trying to get things running on their machine or are evaluating whether it's worth taking the time to try, so I tried to make the example terse but still be batteries-included.

Main changes:
* I explicitly listed setup commands
* I moved service startup instructions in front of connection creation instructions
* I refined descriptions of the task development steps,
* I tried to use phrasing consistent with other documentation pages,
* I included links to help readers discover a convenient workflow (eg that updates to DAGs in code on the host machine (but still in the `dags/` dir) would be discovered after a delay without needing to restart any services).

**Review by potiuk — APPROVED** [review]
Love the attention to details !

**Comment by potiuk** [comment]
Thanks @MattTriano  -> let's see how the CI likes it :)

**Comment by MattTriano** [comment]
```
 ------------------------------ Error   1 --------------------
 WARNING: Block quote ends without a blank line; unexpected unindent.

File path: apache-airflow/tutorial.rst (548)

 545 | And from the last line in the definition of the ``Etl`` DAG, we see:
 546 | 
 547 |   >> get_data() >> merge_data()
>548 | * the ``merge_data()`` task depends on the ``get_data()`` task,
------------------------------ Error   2 --------------------
 ERROR: Unknown target name: "brief delay<https://airflow.apache.org/docs/apache-airflow/stable/configurations-ref.html#dag-dir-list-interval>".
```

These errors caused the `Build docs` CI job to fail. Technically it also threw a spelling error, but the only error traceback from the `Output of spelling check: apache-airflow` section pointed to line 548 citing the lack of a blank line or unindentation. I'm confident I handled that correctly in my latest commit.

For Error 2, I think the issue was the lack of whitespace between the text and the bracketed-URI, but I'm not too familiar with the reStructuredText spec, so it's possible that I'm wrong and wasting more compute.

**Comment by potiuk** [comment]
> For Error 2, I think the issue was the lack of whitespace between the text and the bracketed-URI, but I'm not too familiar with the reStructuredText spec, so it's possible that I'm wrong and wasting more compute.

You can run it locally (see the error output - it explains what commands you can run to build the docs locally)

**Comment by MattTriano** [comment]
What a `./breeze` (...`build-docs -- --spellcheck-only --package-filter apache-airflow`) that was.

**Comment by potiuk** [comment]
Thanks! Really cool!

**Pull request merged**